int8 type
int16 rate
int8 inUse
